### PR TITLE
feat(CWFCONF-3703): Add support for hotfix and prerelease tags

### PR DESCRIPTION
**Pull request by Richard-Filo**
Fixes: https://github.com/ktdreyer/errata-tool-ansible/issues/261

### CODE REVIEW

**Comment by Richard-Filo** [comment]
@ktdreyer Could you take a look on the changes?
Can I run black on all the files to fix the format?
Should I try to create some integration tests as well?

**Comment by ktdreyer** [comment]
I'm going to merge this now since the integration tests pass. Thank you!

**Review by ktdreyer — APPROVED** [review]
None of these issues are critical, and we can fix them separately.

As far as black goes - I'm not opposed do doing it in a separate PR. I disagree that single quotes are less preferable to double-quotes, but otherwise it seems ok.

For integration tests, sure, you can add those.

File: library/errata_tool_cdn_repo.py [hunk 1]
```diff
@@ -530,12 +639,20 @@ def ensure_packages_tags(client, name, check_mode, packages):
 # The end result should match the format of the module
 # params, so refer to the examples in the module docs above.
 def tag_name_or_variant_dict(tag_name, tag_info):
```
[the remark this answers is not in the document]
Comment (Richard-Filo) [reply library/errata_tool_cdn_repo.py:641]: I'll try to fix it in the next PR. Thanks

File: library/errata_tool_cdn_repo.py [hunk 2]
```diff
@@ -234,23 +268,27 @@ def get_package_tags(client, name):
     page_number = 0
     elements = []
     found = []
-    while (page_number == 0 or found == PAGE_SIZE):
+    while (page_number == 0 or len(found) == PAGE_SIZE):
```
[the remark this answers is not in the document]
Comment (Richard-Filo) [reply library/errata_tool_cdn_repo.py:271]: No problem!

File: library/errata_tool_cdn_repo.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Richard-Filo) [reply library/errata_tool_cdn_repo.py:641]: Fixed here: https://github.com/ktdreyer/errata-tool-ansible/pull/316

File: library/errata_tool_cdn_repo.py [hunk 3]
```diff
@@ -175,9 +205,11 @@ def normalize_packages(packages):
...
                           string or a dict.
     :returns: A dict of packages. Each key is a package name. Each value is a
-              dict of tags. Each tag dictionary is empty (to indicate no
-              variant restrictions), or has a "variant" key (to indicate a
-              variant restriction).
+              dict of tags. Each tag dictionary may have a "for_hotfix" key
+              (to indicate if it is for hotfix), a "for_prerelease" key
+              (to indicate if it is for prerelease). If there is no "variant"
+              key (to indicate no variant restrictions), or has a "variant"
+              key (to indicate a variant restriction), .
```
[the remark this answers is not in the document]
Comment (Richard-Filo) [reply library/errata_tool_cdn_repo.py:212]: Fixed here: https://github.com/ktdreyer/errata-tool-ansible/pull/316

**Comment by ktdreyer** [comment]
[GitHub Actions](https://github.com/ktdreyer/errata-tool-ansible/actions/runs/7024070712) uploaded this to [Ansible Galaxy](https://galaxy.ansible.com/ui/repo/published/ktdreyer/errata_tool_ansible/) as version `0.0.331`. Thanks!

**Pull request merged**